net/tshttpproxy/tshttpproxy_synology.go

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) 2022 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+//go:build linux
+// +build linux
+
+package tshttpproxy
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"net/url"
+	"os"
+	"sync"
+	"time"
+
+	"tailscale.com/util/lineread"
+	"tailscale.com/version/distro"
+)
+
+func init() {
+	if distro.Get() == distro.Synology {
+		sysProxyFromEnv = synologyProxyFromConfigCached
+	}
+}
+
+var cachedProxy struct {
+	sync.Mutex
+	val        *url.URL
+	lastUpdate time.Time
+}
+
+func synologyProxyFromConfigCached(req *http.Request) (*url.URL, error) {
+	cachedProxy.Lock()
+	defer cachedProxy.Unlock()
+
+	if time.Since(cachedProxy.lastUpdate) < time.Minute {
+		return cachedProxy.val, nil
+	}
+
+	defer func() {
+		cachedProxy.lastUpdate = time.Now()
+	}()
```

Comment:
is this leftover debugging where you had log prints in here earlier?

just assign this on line 51 without a defer?

Reply:
Embarassingly, it's mental damage from too much time in raii. Anyway, defer removed.